src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -56,9 +56,37 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             catch
             {
                 string message = String.Format(CultureInfo.CurrentCulture,
-                    "The account credentials for '{0}' are incorrect.", account.Credentials.AccountName);
+                    "Invalid storage account '{0}'. Please make sure your credentials are correct.",
+                    account.Credentials.AccountName);
                 throw new InvalidOperationException(message);
             }
+
+            if (isPrimaryAccount)
+            {
+                // Primary storage accounts require Queues
+                var queueClient = account.CreateQueueClient();
+                var queue = queueClient.GetQueueReference("name");
+                try
+                {
+                    await queue.ExistsAsync(cancellationToken);
+                    _primaryCredentials = account.Credentials;
+                }
+                catch (OperationCanceledException)
+                {
+                    throw;
+                }
+                catch (StorageException exception)
+                {
+                    if (exception.Message.Contains("The remote name could not be resolved"))
```

Comment:
This is pretty fragile. Is there an inner exception in those cases (potentially a `WebException`) we can inspect?